src/operator.rs

```diff
@@ -151,6 +151,13 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         let usize = PrimValKind::from_uint_size(self.memory.pointer_size());
         let isize = PrimValKind::from_int_size(self.memory.pointer_size());
         if !left_kind.is_float() && !right_kind.is_float() {
+            if (!left.is_bytes() && !right.is_bytes()) && self.frame().const_env() {
+                if left.is_ptr() && right.is_ptr() {
+                    return Err(EvalError::NotConst("Comparing pointers".to_string()));
```

Comment:
"Comparing" sounds wrong here... this could also be any other operation, couldn't it?